### PR TITLE
LPS-180641 Will cover Expando Column Constants

### PR DESCRIPTION
**Pull request by wesleyroberts**
Will cover the Expando Column Constants present in expandoColumn.json, used by the Site Initializer.
I and @joaodamasceno2001  worked together to fix this task.

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 1]
```diff
@@ -4362,14 +4374,61 @@ private Map<String, String> _getClassNameIdStringUtilReplaceValues() {
...
+			return JSONUtil.toDoubleArray(jsonObject.getJSONArray("defaultValue"));
+		}
+		else if(jsonObject.getInt("dataType") == ExpandoColumnConstants.FLOAT){
+			return jsonObject.getDouble("defaultValue");
+		}
+		else if(jsonObject.getInt("dataType") == ExpandoColumnConstants.FLOAT_ARRAY){
+			return JSONUtil.toFloatArray(jsonObject.getJSONArray("defaultValue"));
+		}
+		else if (jsonObject.getInt("dataType") == ExpandoColumnConstants.GEOLOCATION) {
+			ExpandoValue value = new ExpandoValueImpl();
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:4400]: this is needed for what?

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 2]
```diff
@@ -4362,14 +4374,61 @@ private Map<String, String> _getClassNameIdStringUtilReplaceValues() {
...
+			Geo geo = new Geo();
+			value.setGeolocationJSONObject(jsonObject.getJSONObject("defaultValue"));
+			geo.setLatitude(value.getGeolocationJSONObject().getDouble("latitude"));
+			geo.setLongitude(value.getGeolocationJSONObject().getDouble("longitude"));
```
Comment (nNilton) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:4404]: can't you use Geo.toDTO for this?

**Review by nNilton — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java (hunk 2 above)
Comment (wesleyroberts) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:4404]: yes, we can use Geo.toDTO for this, thank you for sharing with us, now I know a new way for to do this.

Comment (wesleyroberts) [reply modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java]: Needs to do tests with this change, I did it there, but it doesn't work. 
Cast problem:

`java.lang.ClassCastException: com.liferay.headless.commerce.admin.catalog.dto.v1_0.Geo cannot be cast to com.liferay.portal.kernel.json.JSONObject`

**Comment by nNilton** [comment]
resent:https://github.com/pt-liferay-solutions/liferay-portal/pull/3904

**Pull request closed without merging**